amundsen_application/static/js/ducks/search/reducer.ts

```diff
@@ -72,7 +76,7 @@ export default function reducer(state: SearchReducerState = initialState, action
       // updates search term to reflect action
       return {
         ...state,
-        search_term: (<SearchAllRequest>action).term,
+        search_term: (<SearchAllRequest>action).payload.term,
```

Comment:
If you add an explicit type to the action parameter on line 71, you _should_ be able to remove the cast here. 

In `ducks/user/reducer.ts` my IDE is able to correctly detect the action sub-type within each case block.

Reply:
That was removed in the previous cleanup PR because it's technically incorrect. All actions get passed through reducers, it's just that they don't have any effect if the case isn't covered. We were essentially adding a type definition based on what we knew we'd be processing, and not based on what Redux actually does. I have a ticket filed for looking into what's the best pattern for having type saftey on our actions.